### PR TITLE
chore(deps) add OpenResty 1.11.2.2 to supported versions + bump Travis deps

### PR DESCRIPTION
**Pull request by thibaultcha**
- Add OpenResty `1.11.2.2` to Kong's supported versions
- Bump Travis CI Luarocks supported version to `2.4.2`

This supersedes #1839

### CODE REVIEW

File: spec/helpers.lua [hunk 1]
```diff
@@ -747,6 +747,7 @@ luassert:register("assertion", "formparam", req_form_param,
 local function exec(...)
   local ok, _, stdout, stderr = pl_utils.executeex(...)
   if not ok then
+    print("STDOUT: ", stdout)
```
Comment (Tieske) [inline spec/helpers.lua]: This seems debug code. Is that intentional?

File: kong/meta.lua [hunk 2]
```diff
@@ -18,7 +18,7 @@ return {
   -- third-party dependencies' required version, as they would be specified
   -- to lua-version's `set()` in the form {from, to}
   _DEPENDENCIES = {
-    nginx = {"1.11.2.1"},
+    nginx = {"1.11.2.1", "1.11.2.2"},
```
Comment (Tieske) [inline kong/meta.lua]: 1.11.2.2 is the first one to include the breaking Lua 5.2 features, can't have this without pathcing `execute` in Penlight

**Comment by thibaultcha** [comment]
Please disregard this PR for the time being.

**Comment by thibaultcha** [comment]
(given it isn't marked for review, for instance)

**Pull request closed without merging**